cloudant-client/src/main/java/com/cloudant/client/api/query/Sort.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2017 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.client.api.query;
+
+import com.cloudant.client.internal.query.NamedField;
+import com.google.gson.annotations.SerializedName;
+
+public class Sort extends NamedField {
+
+    public enum Order {
+        /**
+         * ascending
+         */
+        @SerializedName("asc")
```

Comment:
I wonder if this and `Type` could use
`toString → return lowercase`
and
`constructor → construct from lowercase`
rather than `SerializedName`

This is more portable if we ever move away from gson and also allows easy construction of queries/selectors in my code which currently avoids explicitly building the JSON tree.

Reply:
I think we'd need both in any case, otherwise we'd still need a custom JSON adapter to call the lower case constructor for deserialization. The way to avoid that would be to break from the Java convention and use `Sort.asc` and `Sort.desc`. Either way I think your proposed query code only goes one way so can cope with a `toString() {return lowercase;}` that we can easily add.

If we go with `.asc("fieldName")` methods on the index builders, we could do similar statics on the `Sort` class to new up the instances instead of using public constructors. It would be shorter and open the avenue to hiding the enum entirely or replacing the values with strings internally instead. It just depends if we think there is any value to the end user in being able to compare to some known set of options vs getting a string.

As an aside the "lock-in" is an annoyance, but we're always going to need some internal JSON adapters for API models even if we allow free choice of deserializer elsewhere (e.g. for user doc models). It's the only way to guarantee things like `serializeNulls()` or similar are set the way we need and to cope with some of the idiosyncrasies of the Couch API. The Java EE JSON specs may provide an avenue to using binding annotations that can be interpreted by multiple JSON parsers, but at the moment I think there are only Jackson and the RI to choose from and I believe they even have some behaviour differences!

Reply:
Done in 5a3343d and 5e3f9ba respectively for `toString`
I don't think we need the constructor at this point.